# You will need to restart basestation.launch for
# Any changes to take effect
teleop:
  ra_names: ["joint_a", "joint_b", "joint_c", "joint_de_pitch", "joint_de_roll", "allen_key", "gripper"]
  ra_controls:
    joint_a:
      multiplier: -1
      xbox_index: "left_x"
    joint_b:
      multiplier: 1
      xbox_index: "left_y"
    joint_c:
      multiplier: 1
      xbox_index: "right_y"
    joint_de_pitch:
      multiplier: -1
      xbox_index_pos: "right_trigger"
      xbox_index_neg: "left_trigger"
    joint_de_roll:
      multiplier: 1
      xbox_index_pos: "right_bumper"
      xbox_index_neg: "left_bumper"
    allen_key:
      multiplier: 1
      xbox_index_pos: "y"
      xbox_index_neg: "a"
    gripper:
      multiplier: 1
      xbox_index_pos: "b"
      xbox_index_neg: "x"

  sa_controls:
    sa_x:
      multiplier: 1
      xbox_index: "left_y"
    sa_y:
      multiplier: -1
      xbox_index: "left_x"
    sa_z:
      multiplier: 1
      xbox_index: "right_y"
    sampler:
      multiplier: 1
      xbox_index_pos: "right_trigger"
      xbox_index_neg: "left_trigger"
    sensor_actuator:
      multiplier: 1
      xbox_index_pos: "right_bumper"
      xbox_index_neg: "left_bumper"
    cache:
      multiplier: 1
      xbox_index_pos: "right_bumper"
      xbox_index_neg: "left_bumper"

  drive_controls:
    forward_back:
      multiplier: 1
    left_right:
      # Can be disabled here since left_right axis sucks
      enabled: true
      multiplier: 1
    twist:
      multiplier: 0.6
    tilt:
      multiplier: 1
    pan:
      multiplier: 1

  xbox_mappings:
    left_x: 0
    left_y: 1
    right_x: 2
    right_y: 3
    a: 0
    b: 1
    x: 2
    y: 3
    left_bumper: 4
    right_bumper: 5
    left_trigger: 6
    right_trigger: 7
    back: 8
    forward: 9
    left_stick_click: 10
    right_stick_click: 11
    dpad_up: 12
    dpad_down: 13
    dpad_left: 14
    dpad_right: 15
    home: 16

  joystick_mappings:
    left_right: 0
    forward_back: 1
    twist: 2
    dampen: 3
    pan: 4
    tilt: 5

  mast_gimbal_power:
    rotation_pwr: 1.0
    up_down_pwr: 0.75

  # Thresholds in radians
  # Euclidean error is the norm of all the joint position errors
  euclidean_error_threshold: 3.14

  joint_error_thresholds:
    joint_a: 1.57
    joint_b: 1.57
    joint_c: 1.57
    joint_d: 1.57
    joint_e: 1.57

  ik_multipliers:
    x: 0.1
    y: 0.1
    z: 0.1